src/nplinker/genomics/bgc.py

```diff
@@ -201,6 +203,36 @@ def to_dict(self) -> dict[str, Any]:
             "antismash_region": self.antismash_region,
         }
 
+    @staticmethod
+    def to_string(value: Any) -> str:
+        """Convert various types of values to a string.
+
+        Args:
+            value: The value to be converted to a string.
+                Can be a list, tuple, set, dict, or any other type.
+
+        Returns:
+            A string representation of the input value.
+        """
+        # Convert list, tuple, set to comma-separated string
+        if isinstance(value, (list, tuple, set)):
+            value = ", ".join(map(str, value))
+        # Convert dict to comma-separated string
+        elif isinstance(value, dict):
+            value = ", ".join([f"{k}:{v}" for k, v in value.items()])
+        # Convert anything else to string
+        else:
+            value = str(value) if value else ""
+        return value
+
+    def to_tabular(self) -> list:
```

Comment:
It's better to return a `dict` instead of a `list`, so that you have both the keys and values, making the use easier.
Otherwise, you have to call `to_dict()` to get the keys, and worry about the order of keys or values.